src/coffee/services/base.coffee

```diff
@@ -556,7 +556,14 @@ class BaseService
 
       # TODO: check other possible acceptable codes (304, ...)
       if 200 <= response.statusCode < 300
-        resolve _.extend responseJson,
+        # FIXME: find a better solution
+        if @constructor.name is 'GraphQLService' and not body.data
+          graphqlError = new GraphQLError 'GraphQL error', _.extend responseJson, body,
+            statusCode: 400
+            originalRequest: originalRequest
+          return reject graphqlError
```

Comment:
hmm.. it's a good point. It's definitely not the part of the spec since it does not cover an HTTP interface. One can still get most of the response data even in presence of errors, so in thin case I would still  return 200. Some errors happen much earlier though, like query validation errors. I think we mat return 400 in these cases.